UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -276,24 +276,44 @@ private static MatrixInfo CreateMatrixInfoFromMatrix(Matrix matrix, int id)
 	}
 
 	/// Finds first matrix that is not empty at given world pos
-	public static MatrixInfo AtPoint(Vector3Int worldPos, bool isServer)
+	public static MatrixInfo AtPoint(Vector3Int worldPos, bool isServer, MatrixInfo possibleMatrix = null)
 	{
+		//for performance, this is just a suggestion on which matrix we believe this point could be in, not always correct
+		if (possibleMatrix != null && IsInMatrix(worldPos, isServer, possibleMatrix))
```

Comment:
```suggestion
		if ((possibleMatrix is null) == false && IsInMatrix(worldPos, isServer, possibleMatrix))
```

Is that null check costly? Might be able to get away with this change.

Reply:
ah wait nvm, isnt a monobehaviour

Reply:
does ``possibleMatrix  is not null`` work? Other options are ``possibleMatrix is {}`` which is uglier but should also work, otherwise ``possibleMatrix is object``

Reply:
> does `possibleMatrix is not null` work? Other options are `possibleMatrix is {}` which is uglier but should also work, otherwise `possibleMatrix is object`

If we have c#9 then you could do `possibleMatrix is not null`. 

Though my comment is irreverent as MatrixInfo isnt a unity object so the null check should be as performant as it can get.